tests/v2/validation/rbac/rbac.go

```diff
@@ -84,3 +100,61 @@ func createProject(client *rancher.Client, clusterID string) (createProject *man
 	return createProject, err
 
 }
+
+func getPSALabels(response *v1.SteveAPIObject, actualLabels map[string]string) map[string]string {
+	expectedLabels := map[string]string{}
+
+	for label := range response.Labels {
+		if _, found := actualLabels[label]; found {
+			expectedLabels[label] = actualLabels[label]
+		}
+	}
+	return expectedLabels
+
+}
+
+func createDeployment(steveclient *v1.Client, client *rancher.Client, clusterID string, containerName string, image string, namespaceName string) (*v1.SteveAPIObject, error) {
+
+	deploymentName := namegen.AppendRandomString("")
+	containerTemplate := workloads.NewContainer(containerName, image, coreV1.PullAlways, []coreV1.VolumeMount{}, []coreV1.EnvFromSource{})
+	matchLabels := map[string]string{}
+	matchLabels["workload.user.cattle.io/workloadselector"] = fmt.Sprintf("apps.deployment-%v-%v", namespaceName, deploymentName)
+
+	podTemplate := workloads.NewPodTemplate([]coreV1.Container{containerTemplate}, []coreV1.Volume{}, []coreV1.LocalObjectReference{}, matchLabels)
+	deployment := workloads.NewDeploymentTemplate(deploymentName, namespaceName, podTemplate, matchLabels)
+
+	deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).Create(deployment)
+	if err != nil {
+		return nil, err
+	}
+	err = kwait.Poll(5*time.Second, 5*time.Minute, func() (done bool, err error) {
+		steveclient, err := client.Steve.ProxyDownstream(clusterID)
+		if err != nil {
+			return false, nil
+		}
+		deploymentResp, err := steveclient.SteveType(workloads.DeploymentSteveType).ByID(deployment.Namespace + "/" + deployment.Name)
+		if err != nil {
+			return false, nil
+		}
+		deployment := &appv1.Deployment{}
+		err = v1.ConvertToK8sType(deploymentResp.JSONResp, deployment)
+		if err != nil {
+			return false, nil
+		}
+		status := deployment.Status.Conditions
+		for _, statusCondition := range status {
+			if strings.Contains(statusCondition.Message, "forbidden") {
+				err = errors.New(statusCondition.Message)
+				return false, err
+			}
+		}
+
+		if *deployment.Spec.Replicas == deployment.Status.AvailableReplicas {
+			return true, nil
+		}
```

Comment:
Another nit, but it seems like this function is not just a create but `createAndWait` or something like that (please name as you want, don't have a strong opinion), but I believe renaming according to the logic can make the helper function usage clearer